collect_app/src/main/java/org/odk/collect/android/tasks/sms/SmsService.java

```diff
@@ -376,4 +377,50 @@ private boolean isFormAutoDeleteEnabled(String jrFormId, boolean isFormAutoDelet
         }
         return autoDelete == null ? isFormAutoDeleteOptionEnabled : Boolean.valueOf(autoDelete);
     }
+
+     public static String getDisplaySubtext(SmsEvent event, Context context) {
+        Date date = event.getLastUpdated();
+        SmsProgress progress = event.getProgress();
+
+        if (event.getStatus() == null) {
+            return null;
+        }
+
+        if (event.getProgress().getPercentage() == 100 && event.getStatus().equals(MessageStatus.Sent)) {
```

Comment:
Just noticed this mutation. If the method is called `getDisplaySubtext` it should really not mutate anything. Perhaps this can happen before the method is called? It might also help to include a comment -- I'm not clear why if the progress is 100% and the event status is "sent" the event status gets changed to "sending."

Reply:
Okay, I am gonna be a bit more descriptive. This is something I think that I might need to refactor.  What's happening is I am utilizing the `SmsStatus` to determine what message should be shown to the user and once the entire submission hasn't been completed then it should transform the state to `Sending`. Because the `SmsStatus` is being mapped from the `BroadcastReceiver` results it was being set as `Sent` so I need to restructure the logic so it's clearer.